### PR TITLE
Spring MVC

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

**Review by viaviaj — APPROVED** [review]
Well done!)

File: src/main/java/intro/config/WebConfig.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
+import org.springframework.web.servlet.view.JstlView;
+
+@Configuration
+@EnableWebMvc
+@ComponentScan(basePackages = {"intro.controller"})
+public class WebConfig {
+    @Bean
+    public InternalResourceViewResolver getResolver() {
```
Comment (Sofasmile) [inline src/main/java/intro/config/WebConfig.java]: let's remove this bean

File: src/main/java/intro/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @GetMapping("/inject")
+    public void injectUsers() {
+        userService.add(new User("test@gmail.com", "123"));
+        userService.add(new User("one_more_test@gmail.com", "321"));
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        UserResponseDto userDto = mapUserToDto(userService.getById(userId));
+        return userDto;
```
Comment (Sofasmile) [inline src/main/java/intro/controller/UserController.java]: better:
```suggestion
        return mapUserToDto(userService.getById(userId));
```
or 
```suggestion
        User user = userService.getById(userId);
        return mapUserToDto(user);
```

File: src/main/java/intro/config/WebConfig.java (hunk 1 above)
Comment (julia-shvabovich) [reply src/main/java/intro/config/WebConfig.java]: done

File: src/main/java/intro/dao/UserDaoImpl.java [hunk 3]
```diff
@@ -42,4 +43,16 @@ public void add(User user) {
             return users.getResultList();
         }
     }
+
+    @Override
+    public Optional<User> getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<User> query = session.createQuery("from User "
```
Comment (RomanDubovskyi) [inline src/main/java/intro/dao/UserDaoImpl.java]: let's use session.get() instead that will make code shorter

File: src/main/java/intro/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        return mapUserToDto(userService.getById(userId));
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        List<UserResponseDto> userDtos = userService.listUsers().stream()
```
Comment (RomanDubovskyi) [inline src/main/java/intro/controller/UserController.java]:
```suggestion
        return userService.listUsers().stream()
```

File: src/main/java/intro/dao/UserDaoImpl.java (hunk 3 above)
Comment (julia-shvabovich) [reply src/main/java/intro/dao/UserDaoImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**